api/v1alpha1/application_types.go

```diff
@@ -64,6 +65,8 @@ type ApplicationSpec struct {
 	// HTTP and HTTPS.
 	//
 	// Ingresses must be lowercase, contain no spaces, be a non-empty string, and have a hostname/domain separated by a period
+	// They can optionally be suffixed with a plus and name of a custom TLS secret located in the istio-gateways namespace.
```

Comment:
I think we should have a CustomCerts type or turn this one into an interface which accepts both an object and string, although this is a dirty solution as well. Or will this make reconciling very difficult?